### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by kalbek**
## Changes made in this PR:
- Create a class Nameable
   - ```Nameable``` inherits from ```Person``` class
   - Implements ```correct_name``` method
 - ```Person``` class to Nameable
   - ```Person``` class inherits from ```Nameable```
   - Implements``` correct_name``` method
-  Prepared the ```Capitalize Decorator```
   - Inherits from the base Decorator class.   
   - Implements ```correct_name``` with capitalization feature
-  Prepared the ```Trimmer Decorator``` 
   - Inherits from the base Decorator class.   
   - Implements ```correct_name``` with trimming feature

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,10 +1,12 @@
 require 'securerandom'
+require './nameable'
```
Comment (codepantha) [inline person.rb:2]: - [ ] - [ ] Please only use `require` when you need to include external dependencies. However, when bringing in your code from another file in the same project, it is best practice to use `require_relative`.

File: capitalize_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# This class represents Capitalize Decorator
+require './base_decorator'
```
Comment (codepantha) [inline capitalize_decorator.rb:2]: - [ ] Please only use `require` when you need to include external dependencies. However, when bringing in your code from another file in the same project, it is best practice to use `require_relative`.

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @kalbek 👋

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Implemented decorators 📌 
✔️ Linters are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by tugsem — APPROVED** [review]
## STATUS: APPROVED 👍🎯  
Hello @kalbek 👋 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights ⭐ 
✔️ changes impelemented
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**